azurerm/resource_arm_private_dns_srv_record.go

```diff
@@ -0,0 +1,252 @@
+package azurerm
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPrivateDnsSrvRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPrivateDnsSrvRecordCreateUpdate,
+		Read:   resourceArmPrivateDnsSrvRecordRead,
+		Update: resourceArmPrivateDnsSrvRecordCreateUpdate,
+		Delete: resourceArmPrivateDnsSrvRecordDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
```

Comment:
running the tests it appears due to the API bug outlined below that the name is also lower-cased in the API response. Since DNS addresses themselves need to be lower-cased - I hope you don't mind but I'm going to push a commit to add some validation to this resource (and the other Private DNS resources) to ensure that the name field is lower-cased